<launch>
    <!-- 3 ports A,B,C configuration
    <arg name="img_topic_0" default="/port_0/camera_0/image_raw/compressed"/>
	<arg name="img_topic_1" default="/port_0/camera_1/image_raw/compressed"/>
	<arg name="img_topic_2" default="/port_0/camera_2/image_raw/compressed"/>
	<arg name="img_topic_3" default="/port_1/camera_0/image_raw/compressed"/>
	<arg name="img_topic_4" default="/port_1/camera_1/image_raw/compressed"/>
	<arg name="img_topic_5" default="/port_1/camera_2/image_raw/compressed"/>
    <arg name="img_topic_6" default="/port_2/camera_0/image_raw/compressed"/>
	<arg name="img_topic_7" default="/port_2/camera_1/image_raw/compressed"/>
    -->

    <!-- 2 ports A,B configuration -->
	<arg name="img_topic_0" default="/port_0/camera_0/image_raw/compressed"/>
	<arg name="img_topic_1" default="/port_0/camera_1/image_raw/compressed"/>
	<arg name="img_topic_2" default="/port_0/camera_2/image_raw/compressed"/>
	<arg name="img_topic_3" default="/port_0/camera_3/image_raw/compressed"/>
	<arg name="img_topic_4" default="/port_1/camera_0/image_raw/compressed"/>
	<arg name="img_topic_5" default="/port_1/camera_1/image_raw/compressed"/>
    <arg name="img_topic_6" default="/port_1/camera_2/image_raw/compressed"/>
	<arg name="img_topic_7" default="/port_1/camera_3/image_raw/compressed"/>


    <!-- nodes -->
	<node pkg="image_compressed_view" type="image_view_compressed.py" name="viewer_0" output="screen" >
		<param name="img_topic" value="$(arg img_topic_0)"/>
	</node>
	<node pkg="image_compressed_view" type="image_view_compressed.py" name="viewer_1" output="screen" >
		<param name="img_topic" value="$(arg img_topic_1)"/>
	</node>
	<node pkg="image_compressed_view" type="image_view_compressed.py" name="viewer_2" output="screen" >
		<param name="img_topic" value="$(arg img_topic_2)"/>
	</node>
	<node pkg="image_compressed_view" type="image_view_compressed.py" name="viewer_3" output="screen" >
		<param name="img_topic" value="$(arg img_topic_3)"/>
	</node>
    <node pkg="image_compressed_view" type="image_view_compressed.py" name="viewer_4" output="screen" >
		<param name="img_topic" value="$(arg img_topic_4)"/>
	</node>
    <node pkg="image_compressed_view" type="image_view_compressed.py" name="viewer_5" output="screen" >
		<param name="img_topic" value="$(arg img_topic_5)"/>
	</node>
	<node pkg="image_compressed_view" type="image_view_compressed.py" name="viewer_6" output="screen" >
		<param name="img_topic" value="$(arg img_topic_6)"/>
	</node>
	<node pkg="image_compressed_view" type="image_view_compressed.py" name="viewer_7" output="screen" >
		<param name="img_topic" value="$(arg img_topic_7)"/>
	</node>


</launch>
